app/controllers/students_controller.rb

```diff
@@ -0,0 +1,27 @@
+class StudentsController < ApplicationController
+
+  def index
+    @students = Student.all
+  end
+
+  def show
+    @student = Student.find(params[:id])
+  end
+
+  def new
+      @student = Student.new
+  end
+
+  def create
+      @student = Student.new(student_params)
+      @student = Student.sorting_hat @student
+      @student = Student.create!({:name => @student.name, :img_url => @student.img_url, :house => @student.house})
```

Comment:
Is there a reason you're reassigning `name` and `img_url`? `Student.new(student_params)` should already have taken care of that.